api/v1beta1/consts.go

```diff
@@ -142,3 +142,18 @@ const (
 	// UpdatingReason means the resource is being updated.
 	UpdatingReason = "Updating"
 )
+
+const (
+	// CustomHeaderPrefix is the prefix of annotations that enable additional cluster / node pool features.
+	// Whatever follows the prefix will be passed as a header to cluster/node pool creation/update requests.
+	// E.g. add `"infrastructure.cluster.x-k8s.io/custom-header-UseGPUDedicatedVHD": "true"` annotation to
+	// AzureManagedMachinePool CR to enable creating GPU nodes by the node pool.
+	CustomHeaderPrefix = "infrastructure.cluster.x-k8s.io/custom-header-"
+)
+
+const (
+	// LinuxOS is Linux OS value for OSDisk.OSType.
```

Comment:
Does this really belong in the api/ folder? Seems like it would be a better fit for azure/consts.go or defaults.go

Reply:
we actually already have it in defaults.go https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/defaults.go#L47